src/explore-education-statistics-common/src/modules/table-tool/components/__tests__/LocationFiltersForm.test.tsx

```diff
@@ -0,0 +1,298 @@
+import LocationFiltersForm, {
+  LocationFormValues,
+} from '@common/modules/table-tool/components/LocationFiltersForm';
+import { InjectedWizardProps } from '@common/modules/table-tool/components/Wizard';
+import { SubjectMeta } from '@common/services/tableBuilderService';
+import { waitFor } from '@testing-library/dom';
+import { render, screen, within } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import noop from 'lodash/noop';
+import React from 'react';
+
+describe('LocationFiltersForm', () => {
+  const testWizardStepProps: InjectedWizardProps = {
+    currentStep: 1,
+    isActive: true,
+    isLoading: false,
+    stepNumber: 1,
+    setCurrentStep: noop,
+    goToNextStep: noop,
+    goToPreviousStep: noop,
+    shouldScroll: false,
+  };
+
+  const testLocations: SubjectMeta['locations'] = {
+    Country: {
+      hint: '',
+      legend: 'Country',
+      options: [
+        {
+          label: 'Country 1',
+          value: 'country-1',
+        },
+      ],
+    },
+    LocalAuthority: {
+      hint: '',
+      legend: 'Local authority',
+      options: [
+        {
+          label: 'Local authority 1',
+          value: 'local-authority-1',
+        },
+        {
+          label: 'Local authority 2',
+          value: 'local-authority-2',
+        },
+        {
+          label: 'Local authority 3',
+          value: 'local-authority-3',
+        },
+      ],
+    },
+    Region: {
+      hint: '',
+      legend: 'Region',
+      options: [
+        {
+          label: 'Region 1',
+          value: 'region-1',
+        },
+        {
+          label: 'Region 2',
+          value: 'region-2',
+        },
+      ],
+    },
+  };
+
+  const testSingleLocation: SubjectMeta['locations'] = {
+    Country: {
+      hint: '',
+      legend: 'Country',
+      options: [
+        {
+          label: 'Country 1',
+          value: 'country-1',
+        },
+      ],
+    },
+  };
+
+  test('renders location group options correctly', () => {
+    render(
+      <LocationFiltersForm
+        {...testWizardStepProps}
+        options={testLocations}
+        onSubmit={noop}
+      />,
+    );
+
+    expect(
+      within(
+        screen.getByRole('group', {
+          name: 'Country',
+          hidden: true,
+        }),
+      ).getAllByRole('checkbox', {
+        hidden: true,
+      }),
+    ).toHaveLength(1);
+
+    expect(
+      within(
+        screen.getByRole('group', {
+          name: 'Local authority',
+          hidden: true,
+        }),
+      ).getAllByRole('checkbox', {
+        hidden: true,
+      }),
+    ).toHaveLength(3);
+
+    expect(
+      within(
+        screen.getByRole('group', {
+          name: 'Region',
+          hidden: true,
+        }),
+      ).getAllByRole('checkbox', {
+        hidden: true,
+      }),
+    ).toHaveLength(2);
```

Comment:
Would suggest adding some extra checks to see if the options themselves are correctly labelled and ordered for each group. We should be precise about which group each option is in by using `within` as well.